modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/compaction/Compactor.java

```diff
@@ -485,4 +500,52 @@ private DeltaFileForCompaction(
             this.deltaFilePageStoreIo = deltaFilePageStoreIo;
         }
     }
+
+    /**
+     * Pauses the compactor until it is resumed or compactor or stopped. It is expected that this method will not be called multiple times
+     * in parallel and subsequent calls will strictly be calls after {@link #resume}.
+     */
+    public void pause() {
+        synchronized (pauseMux) {
+            assert !paused : "It is expected that a further pause will only occur after resume";
+
+            paused = true;
+        }
+    }
+
+    /** Resumes the compactor if it was paused. It is expected that this method will not be called multiple times in parallel. */
+    public void resume() {
+        synchronized (pauseMux) {
+            if (paused) {
+                paused = false;
+
+                pauseMux.notifyAll();
+            }
+        }
+    }
+
+    /** Must be called before each IO operation to provide other IO components with resources. */
+    private void pauseCompactionIfNeeded() {
+        boolean interrupted = false;
+
+        synchronized (pauseMux) {
+            while (paused) {
+                blockingSectionBegin();
+
+                try {
+                    pauseMux.wait();
+                } catch (InterruptedException e) {
```

Comment:
I think if you got interrupted, then you need to re-throw this exception, not just log it, because I would expect the compaction to stop altogether